File: rust-arroyo/src/processing/strategies/noop.rs
use std::time::Duration;

use crate::types::Message;

use super::{CommitRequest, ProcessingStrategy, StrategyError, SubmitError};

/// Noop strategy that takes a message and does nothing.
///
/// This can be useful when you do not care to commit an offset.
pub struct Noop {}

impl<TPayload> ProcessingStrategy<TPayload> for Noop {
    fn poll(&mut self) -> Result<Option<CommitRequest>, StrategyError> {
        Ok(None)
    }
    fn submit(&mut self, _message: Message<TPayload>) -> Result<(), SubmitError<TPayload>> {
        Ok(())
    }
    fn terminate(&mut self) {}
    fn join(&mut self, _timeout: Option<Duration>) -> Result<Option<CommitRequest>, StrategyError> {
        Ok(None)
    }
}

File: arroyo/processing/strategies/noop.py
from __future__ import annotations

from typing import Optional, Union

from arroyo.processing.strategies.abstract import ProcessingStrategy
from arroyo.types import FilteredPayload, Message, TStrategyPayload


class Noop(
    ProcessingStrategy[Union[FilteredPayload, object]],
):
    """
    Noop strategy that takes a message and does nothing.
    """

    def __init__(self) -> None:
        pass

    def submit(
        self, message: Message[Union[FilteredPayload, TStrategyPayload]]
    ) -> None:
        pass

    def poll(self) -> None:
        pass

    def join(self, timeout: Optional[float] = None) -> None:
        pass

    def close(self) -> None:
        pass

    def terminate(self) -> None:
        pass
